### PR TITLE
Move year slider to filtering panel and make it a range slider

### LINKED ISSUE #117: Move "discovered before" control to Filtering

### LINKED ISSUE #125: Convert discovery year to a slider

### PR DESCRIPTION
**Pull request by waxlamp**
Closes #117 
Closes #125

### CODE REVIEW

**Comment by waxlamp** [comment]
I am getting the following error when I manipulate the year range slider:

```
[mobx] `observableArray.sort()` will not update the array in place. Use `observableArray.slice().sort()` to supress this warning and perform the operation on a copy, or `observableArray.replace(observableArray.slice().sort())` to sort & update in place
```

@sgratzl, do you know what's causing this? I can't find the immediate problem.

File: src/components/graph-vis/filters.js [hunk 1]
```diff
@@ -16,6 +18,21 @@ class Filters extends React.Component {
   render() {
     const store = this.context;
     return (<Grid>
+      { /* dataset specific */}
+      {store.yearRange && <RangeSliderControlComponent
+        value={store.year}
```
Comment (sgratzl) [inline src/components/graph-vis/filters.js]:
```suggestion
        value={store.year.slice()}
```

the Slider component internally calls `.sort` on its given values. Thus, the mobx warning.

File: src/store/index.js [hunk 2]
```diff
@@ -153,10 +153,10 @@ export class ApplicationStore {
 
     @computed
     get filterDiscoveryYear() {
-        const cutoff = this.year;
+        const [low, high] = this.year;
         return (node) => {
-            const year = node.discovery;
-            return (year == null && cutoff === 2016) || (year != null && year <= cutoff);
+            const discovery = node.discovery;
```
Comment (sgratzl) [inline src/store/index.js]: instead of the 2016, could you use `this.yearRange[1]`?

Comment (waxlamp) [reply src/store/index.js]: Good catch, thanks. See 99cc9f2017a33aec09c5a8b71969ac9941c7fce1.

**Pull request merged**